kore/src/Kore/Step/Rule/Simplify.hs

```diff
@@ -51,15 +55,17 @@ import Kore.Step.Simplification.Simplify
     , SimplifierVariable
     )
 
-simplifyOnePathRuleLhs
+simplifyGoalRuleLhs
     ::  ( MonadSimplify simplifier
         , SimplifierVariable variable
+        , Coercible (RulePattern variable) rule
+        , Coercible rule (RulePattern variable)
         )
-    => OnePathRule variable
-    -> simplifier (MultiAnd (OnePathRule variable))
-simplifyOnePathRuleLhs (OnePathRule rule) = do
-    simplifiedList <- simplifyRuleLhs rule
-    return (fmap OnePathRule simplifiedList)
+    => rule
+    -> simplifier (MultiAnd rule)
+simplifyGoalRuleLhs rule = do
```

Comment:
This is another one I would probably prefer to see lifted into the class.

Reply:
I lifted both the above, but each to its own class and included them both into the `Claim` type.  Please take another look.